packages/settings-library/src/settings_library/base.py

```diff
@@ -1,74 +1,111 @@
-import logging
-import os
 from functools import cached_property
-from typing import List, Tuple, Type
+from typing import Sequence, get_args
 
-from pydantic import BaseSettings, Extra, SecretStr, ValidationError
+from pydantic import BaseConfig, BaseSettings, Extra, ValidationError, validator
+from pydantic.error_wrappers import ErrorList, ErrorWrapper
+from pydantic.fields import ModelField, Undefined
+from pydantic.types import SecretStr
 
-logger = logging.getLogger(__name__)
+
+class DefaultFromEnvFactoryError(ValidationError):
+    ...
+
+
+def create_settings_from_env(field):
+    # Keeps a reference of field but MUST nothing should be modified there
+    # cannot pass only field.type_ because @prepare_field still not resolved!
+
+    def _default_factory():
+        """Creates default from sub-settings or None (if nullable)"""
+        field_settings_cls = field.type_
+        try:
+            return field_settings_cls()
+
+        except ValidationError as err:
+            if field.allow_none:
+                return None
+
+            def _prepend_field_name(ee: ErrorList):
+                if isinstance(ee, ErrorWrapper):
+                    return ErrorWrapper(ee.exc, (field.name,) + ee.loc_tuple())
+                assert isinstance(ee, Sequence)  # nosec
+                return [_prepend_field_name(e) for e in ee]
+
+            raise DefaultFromEnvFactoryError(
+                errors=_prepend_field_name(err.raw_errors),  # type: ignore
+                model=err.model,
+                # FIXME: model = shall be the parent settings?? but I dont find how retrieve it from the field
```

Comment:
is that still current?

Reply:
yes. I do not now how to solve it. The result is that the error will have a wrong reference to the parent model :-(

The problem is that I need to find the parent model and the only info I have is the child ``FieldModel``. I do not know how to achieve this. Any hints?